### ISSUE TITLE
Unexpected token when parsing stateless component with flow annotated return type

### ISSUE THREAD

**Issue by meleyal** [yannickcr/eslint-plugin-react#390]
I'm not sure if this is the right place for this issue, but only eslint complains about the following code, both babel and flow compile without errors / warnings.

```js
export const PageList = ({ pages, width, height }: Props): ReactElement => {
  ...
}
```

results in:

```shell
20:61  error  Parsing error: Unexpected token
```

**Comment by yannickcr** [comment]
It seems you're using the ESLint default parser [Espree](https://github.com/eslint/espree), but Espree does not support flow annotations (https://github.com/eslint/espree/issues/189).

You'll need to use [babel-eslint](https://github.com/babel/babel-eslint) for this.

**Comment by meleyal** [comment]
I'm actually using `"babel-eslint": "^4.1.6"` already.

**Comment by meleyal** [comment]
I've also tried `"parser": "babel-eslint"` in `.eslintrc`.

**Comment by meleyal** [comment]
Any other ideas on what could be causing the error?

**Comment by yannickcr** [comment]
I cannot reproduce the error with `eslint@2.0.0-beta.1` and `babel-eslint@4.1.6`.

I'm pretty sure this is not related to `eslint-plugin-react` at all since the parse error come from ESLint.

But if you can provide me more information (eslint/babel-eslint versions, `.eslintrc` file and the full source file) maybe I could find the problem.

**Comment by meleyal** [comment]
`eslint@2.0.0-beta.1` + `babel-eslint@4.1.6`

`.eslintrc`:
```json
{
  "parser": "babel-eslint",
  "extends": "rackt",
  "env": {
    "mocha": true
  },
  "rules": {
    "strict": 0,
    "new-cap": 0,
    "comma-dangle": [1, "always-multiline"],
    "react/jsx-uses-react": 1,
    "react/wrap-multilines": 2,
    "flow-vars/define-flow-type": 1,
    "flow-vars/use-flow-type": 1
  },
  "plugins": [
    "react",
    "flow-vars"
  ]
}
```

`MyComponent.js`:
```js
// @flow

import React from 'react'
import { Page } from '../models/Page'

type Props = {
  page: Page,
  width: number,
  height: number
}

export const MyComponent = (props: Props): ReactElement => {
  ...

  return (
    <img key={ref} src={imageUrl} style={style} />
  )
}
```

**Comment by DaleJefferson** [comment]
@meleyal I can reproduce this issue without eslint-plugin-react I think we need to raise this in ESLint or babel-eslint

This is the simplest version that throws the Parsing error: Unexpected token error

```
const add = (a: number, b: number): number => a + b;
```

**Comment by meleyal** [comment]
Upgrading to `babel-eslint@5.0.0-beta6` fixes the issue: https://github.com/babel/babel-eslint/issues/237
